infra/main.bicep

```diff
@@ -23,36 +16,140 @@ param location string
 ])
 param useMonitoring string
 
-@description('API Center instance name to connect. (e.g. apic-linter1234, etc.) If not provided, a new API Center instance will be provisioned.')
-param apicName string
+// Optional parameters to override the default azd resource naming conventions. Update the main.parameters.json file to provide values. e.g.,:
+// "resourceGroupName": {
+//      "value": "myGroupName"
+// }
+param functionName string = ''
+param appServicePlanName string = ''
+param storageAccountName string = ''
+param apiCenterName string = ''
```

Comment:
It would be great if we remove this default value setting part, so that devs will be prompted to enter the name, unless they manually run `azd env set`.